spec/bashcov/detective_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Bashcov::Detective do
+  let(:detective) { described_class.new("bash") }
+
+  describe "#shellscript_shebang_line?" do
+    let(:whitespace)  { ["", " ", "   ", "\t"] }
+    let(:bindirs)     { %w[bin usr/bin usr/local/bin].map { |d| "/#{d}/" } }
+    let(:prefixes)    { [""] + bindirs }
+    let(:shells)      { %w[sh bash] }
+    let(:notshells)   { %w[python3 ruby emacs 1.2.3 --hmm] }
+    let(:envs)        { combine(["env", "env -S"]) }
+    let(:notenvs)     { combine(["false", "hello --world"]) }
+    let(:valid)       { make_shebangs(shells) }
+    let(:invalid) do
+      ["", "\n", "\t\n", "foo"] + make_shebangs(notshells) + shebangify(with_executors(combine(shells), executors: notenvs))
+    end
+
+    def combine(candidates)
+      whitespace.product(prefixes).map(&:join).product(candidates).map(&:join)
+    end
+
+    def with_executors(interpreters, executors: envs)
+      interpreters.flat_map { |i| executors.map { |e| [e, i].join(" ") } }
+    end
+
+    def combine_with_executors(candidates, executors: envs)
+      combine(candidates).yield_self do |base|
+        base + with_executors(base, executors: executors)
+      end
+    end
+
+    def shebangify(candidates)
+      candidates.map { |c| "#!#{c}" }
+    end
+
+    def make_shebangs(candidates, executors: envs)
+      shebangify(combine_with_executors(candidates, executors: executors))
+    end
+
+    it "returns true for shell interpreters" do
+      aggregate_failures "valid shebang handling" do
```

Comment:
TIL about [`aggregate_failures`](https://www.rubydoc.info/github/rspec/rspec-expectations/RSpec%2FMatchers%3Aaggregate_failures), nice :)

Reply:
I see an application for `aggregate_failures` in the runner spec as well; PR incoming :)